### ISSUE TITLE
Inconsistent results between estimator and graph/session

### ISSUE THREAD

**Issue by alee0618** [tensorflow/tensorflow#23957]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10.0.17134
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below):1.12.0
- Python version:3.5
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:9.0
- GPU model and memory: GTX1070, 8Gb

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**
I implemented the same model (2-layer fully-connected neural network + batch norm to reduce initialization effects) using 
1. Low level APIs (graphs and sessions), and
2. High level APIs (estimators)
I tested the two implementation and tested it on the MNIST dataset. The Estimator shows a much slower training and a significantly lower final recognition error. (Estimator: First epoch~92%, Final~96%,  Sess+Graph: First epoch~96%, Final>98%)
**Describe the expected behavior**
The performance should be similar.
**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

The High-Level API
```
import tensorflow as tf
import numpy as np

#Data
from tensorflow.examples.tutorials.mnist import input_data
data = input_data.read_data_sets("/tmp/data/", one_hot = True)
n_inputs = 784
n_classes = 10
'''Input Function'''
train_input_fn = tf.estimator.inputs.numpy_input_fn(
    x={"x":np.array(data.train.images)},
    y=np.array(data.train.labels),
    num_epochs=1,
    shuffle=True)
'''Test Function'''
test_input_fn = tf.estimator.inputs.numpy_input_fn(
    x={"x":np.array(data.test.images)},
    y=np.array(data.test.labels),
    num_epochs=1,
    shuffle=False)

#Training Parameters
params={"lr_init" :0.1,
        "n_neurons":200}


#Model function
def model_fn(features, labels, mode, params):
    '''Model Graph'''
    x = features["x"]
    prediction=tf.layers.dense(x, units=params["n_neurons"], activation=None, use_bias=False)
    prediction=tf.layers.batch_normalization(prediction, training=(mode == tf.estimator.ModeKeys.TRAIN))
    prediction=tf.nn.relu(prediction)
    prediction=tf.layers.dense(prediction, units=n_classes, activation=None)
    
    if(mode==tf.estimator.ModeKeys.PREDICT):
        '''If Evaluation, then simply return predictions'''
        spec = tf.estimator.EstimatorSpec(mode=mode, predictions=prediction)
    else:
        '''If Training or Testing, compute loss and accuracy'''
        loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits=prediction, labels=labels))
        metrics = {"acuracy":tf.metrics.accuracy(tf.argmax(labels, 1), tf.argmax(prediction, 1))}        
        '''Training Operation'''
        with tf.control_dependencies(tf.get_collection(tf.GraphKeys.UPDATE_OPS)):
            optimizer = tf.train.MomentumOptimizer(learning_rate=params["lr_init"], momentum=0.9)
        train_op = optimizer.minimize(loss, global_step=tf.train.get_global_step())        
        '''Specs for Training and Evaluation'''
        spec=tf.estimator.EstimatorSpec(
            mode=mode,
            loss=loss,
            train_op=train_op,
            eval_metric_ops=metrics)
    
    return spec

model=tf.estimator.Estimator(model_fn=model_fn,
                             params=params,
                             model_dir='./estimator_out/')
for i in range(100):
    model.train(input_fn=train_input_fn)
    result = model.evaluate(input_fn=test_input_fn)
    print(result)

```
The Low-Level API:
```
import tensorflow as tf
import numpy as np

#Data
from tensorflow.examples.tutorials.mnist import input_data
mnist = input_data.read_data_sets("/tmp/data/", one_hot = True)
n_inputs=784
n_classes=10

'''Training Parameters'''
params={"lr_init" :0.1,
        "n_neurons":200}
train_batch=128

'''Model graph'''                      
x = tf.placeholder(tf.float32, shape=[None, n_inputs])
y = tf.placeholder(tf.float32, shape=[None, n_classes])
learning_rate = tf.placeholder(tf.float32)
training = tf.placeholder(tf.bool)
prediction=tf.layers.dense(x, units=params["n_neurons"], activation=None, use_bias=False)
prediction=tf.layers.batch_normalization(prediction, training=training)
prediction=tf.nn.relu(prediction)
prediction=tf.layers.dense(prediction, units=n_classes, activation=None)

'''loss and accuracy'''
loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits=prediction, labels=y))
with tf.control_dependencies(tf.get_collection(tf.GraphKeys.UPDATE_OPS)):
    optimizer = tf.train.MomentumOptimizer(learning_rate, momentum=0.9).minimize(loss)
correct = tf.equal(tf.argmax(prediction, 1), tf.argmax(y,1))
accuracy = tf.reduce_mean(tf.cast(correct, 'float'))

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    for epoch in range(100):
        for batch in range(int(mnist.train.num_examples/train_batch)):
            epoch_x, epoch_y=mnist.train.next_batch(train_batch, shuffle=True)
            sess.run(optimizer, feed_dict = {x:epoch_x, y:epoch_y, learning_rate:params["lr_init"], training:True})
        acc = sess.run(accuracy, feed_dict = {x:mnist.test.images, y:mnist.test.labels, learning_rate:0, training:False})
        print('accuracy:',acc)
```
